platform/ios/scripts/generate-style-code.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const fs = require('fs');
+const ejs = require('ejs');
+const spec = require('mapbox-gl-style-spec').latest;
+
+global.camelize = function (str) {
+    return str.replace(/(?:^|-)(.)/g, function (_, x) {
+        return x.toUpperCase();
+    });
+}
+
+global.camelizeWithLeadingLowercase = function (str) {
+    return str.replace(/-(.)/g, function (_, x) {
+        return x.toUpperCase();
+    });
+}
+
+global.isObject = function (property) {
+    switch (property.type) {
+        case 'string':
+            return true;
+        case 'color':
+            return true;
+        case 'array':
+            return true;
+        case 'enum': // (enum will be converted to NSString hence isObject(enum) == true)
+            return true;
+        default:
+            return false;
+    }
+}
+
+global.propertyType = function (property) {
+    switch (property.type) {
+        case 'boolean':
+            return 'BOOL';
+        case 'number':
+            return 'float';
+        case 'string':
+            return 'NSString';
+        case 'enum':
+            return 'NSString';
+        case 'color':
+            return `MGLColor`;
+        case 'array':
+            if (property.length) {
+                return `NSMutableArray`;
+            } else {
+                return `CGVector`;
```

Comment:
There is at least one case where the array represents an offset (iconOffset) which is why I added CGVector but the implementation is mixed up and not ready. Also, the array shouldn't be mutable.